frontend/src/hooks/localLabels.ts

```diff
@@ -34,24 +36,30 @@ export function useLocalLabels(): LocalLabelHook | NotEnabled {
   }
 
   const updateLabel: SetLocalLabel = (alias, newLabel) => {
-    const type = isRandomAlias(alias) ? "random" : "custom";
+    const maskType = isRandomAlias(alias) ? "random" : "custom";
 
     // Replace existing labels for this alias:
     const oldLocalLabel = localLabels.find(
-      (localLabel) => localLabel.id === alias.id && localLabel.type === type
+      (localLabel) =>
+        localLabel.id === alias.id &&
+        (localLabel.mask_type === maskType ||
+          (typeof localLabel.mask_type === "undefined" &&
+            localLabel.type === maskType))
```

Comment:
Fallback for add-on versions without https://github.com/mozilla/fx-private-relay-add-on/pull/344 - can be removed when that is rolled out broadly.